### PR TITLE
Simplify graph_dedup [run_process_replay]

### PR DESCRIPTION
**Pull request by DKormann**

### CODE REVIEW

File: tinygrad/codegen/uops.py [hunk 1]
```diff
@@ -325,25 +325,21 @@ def graph_dedup(self, sink:UOp):
...
-    early_queue = [x for x in all_nodes if early_in_degree[x] == 0]
+    early_queue = [k for k, v in early_in_degree.items() if v == 0]
     replace_nodes: Dict[UOp, UOp] = {}
     while len(early_queue):
       n = early_queue.pop(0)
       if n in replace_nodes: continue
       key = (n.op, n.dtype, tuple(replace_nodes.get(x, x) for x in n.src), n.arg)
-      if found:=self.nodes.get(key): replace_nodes[n] = found
-      else: replace_nodes[n] = self.nodes[key] = UOp(*key)
+      replace_nodes[n] = self.nodes.setdefault(key, UOp(*key))
```
Comment (geohot) [inline tinygrad/codegen/uops.py:342]: Careful, this is probably slower.

Comment (DKormann) [reply tinygrad/codegen/uops.py]: yeah a little

**Comment by DKormann** [comment]
this one should be a speedup btw

**Comment by Qazalin** [comment]
minor benchmark change,
master:
```
***** model forward in  53.18 ms
***** model schedule in  16.89 ms
***** model lower in 2138.58 ms
```
this branch:
```
***** model forward in  54.26 ms
***** model schedule in  17.03 ms
***** model lower in 2139.69 ms
```
I like that all_nodes is gone

**Pull request merged**